PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,31 @@
+(Describe your pull request in 1-2 sentences)
+
+Fix # (optional)
+
+Rationale and detailed description of the changes:
+
+-
+-
+-
+
+Tasks before review:
+
+- [ ] Add tests and examples of any added functionality
+    - [ ] For bugfix pull requests, list any previously-disabled tests that are
+          now passing (and any tests to catch similar bugs elsewhere in the
+          code)
+- [ ] Run `make lint` on the project, ensure it passes
+    - [ ] Run `make pretty-cpp` and `make pretty-python`, if necessary
```

Comment:
I would remove 'if necessary', this can always be done

Reply:
Well no, I often don't run it either because I don't need to or because it makes the formatting worse.

Reply:
Although I admit that's not the best way to deal with bad auto-formatting, since someone else is just going to run `pretty` over it again. Maybe I'll just stick with `clang-format off` where necessary.